costmap_2d/include/costmap_2d/observation.h

```diff
@@ -97,6 +98,7 @@ class Observation
   geometry_msgs::Point origin_;
   pcl::PointCloud<pcl::PointXYZ>* cloud_;
   double obstacle_range_, raytrace_range_;
+  std::vector<bool> within_range_;
```

Comment:
rather than creating a possibly very large bool array, I would suggest that we should instead pass the min_obstacle_height_ and max_obstacle_height_ into the Observation from the ObservationBuffer (as we already do for raytrace_range_ and obstacle_range_)

Reply:
I agree, we can do that